kernel/src/table_properties.rs

```diff
@@ -0,0 +1,299 @@
+#[strum(serialize_all = "camelCase")]
+pub enum CheckpointPolicy {
+    /// classic Delta Lake checkpoints
+    #[default]
+    Classic,
+    /// v2 checkpoints
+    V2,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::expressions::column_name;
+    use std::collections::HashMap;
+
+    #[test]
+    fn known_key_unknown_val() {
+        let properties = HashMap::from([("delta.appendOnly".to_string(), "wack".to_string())]);
+        let table_properties = TableProperties::from(properties.iter());
+        let unknown_properties =
+            HashMap::from([("delta.appendOnly".to_string(), "wack".to_string())]);
+        let expected = TableProperties {
+            unknown_properties,
+            ..Default::default()
+        };
+        assert_eq!(table_properties, expected);
+    }
+
+    #[test]
+    fn allow_unknown_keys() {
+        let properties = [("some_random_unknown_key".to_string(), "test".to_string())];
+        let actual = TableProperties::from(properties.clone().into_iter());
+        let expected = TableProperties {
+            unknown_properties: HashMap::from(properties),
+            ..Default::default()
+        };
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn test_empty_table_properties() {
+        let map: HashMap<String, String> = HashMap::new();
+        let actual = TableProperties::from(map.iter());
+        let default_table_properties = TableProperties::default();
+        assert_eq!(actual, default_table_properties);
+    }
+
+    #[test]
+    fn test_parse_table_properties() {
+        let properties = [
+            ("delta.appendOnly", "true"),
+            ("delta.autoOptimize.optimizeWrite", "true"),
+            ("delta.autoOptimize.autoCompact", "true"),
+            ("delta.checkpointInterval", "101"),
+            ("delta.checkpoint.writeStatsAsJson", "true"),
+            ("delta.checkpoint.writeStatsAsStruct", "true"),
+            ("delta.columnMapping.mode", "id"),
+            ("delta.dataSkippingNumIndexedCols", "-1"),
+            ("delta.dataSkippingStatsColumns", "col1,col2"),
+            ("delta.deletedFileRetentionDuration", "interval 1 second"),
+            ("delta.enableChangeDataFeed", "true"),
+            ("delta.enableDeletionVectors", "true"),
+            ("delta.isolationLevel", "snapshotIsolation"),
+            ("delta.logRetentionDuration", "interval 2 seconds"),
+            ("delta.enableExpiredLogCleanup", "true"),
+            ("delta.randomizeFilePrefixes", "true"),
+            ("delta.randomPrefixLength", "1001"),
+            (
+                "delta.setTransactionRetentionDuration",
+                "interval 60 seconds",
+            ),
+            ("delta.targetFileSize", "1000000000"),
+            ("delta.tuneFileSizesForRewrites", "true"),
+            ("delta.checkpointPolicy", "v2"),
+            ("delta.enableRowTracking", "true"),
+        ];
+        let actual = TableProperties::from(properties.into_iter());
```

Comment:
Ah... this is a pretty good reason to not demand and consume an actual hash map...

Reply:
you mean just for the `into_iter()` API?

Reply:
The current API can consume both `[(&str, &str); _]` and `HashMap<String, String>`. Otherwise these unit tests would involve a lot of "fun" `.to_string()` calls...